### PR TITLE
Remove OPL licensing option.

### PR DESCRIPTION
**Pull request by gmaxwell**
The OPL enables the author to both prohibit modifications
 without their explicit approval, as well as enabling them
 to prohibit distribution in print form.

This is antithetical to the goals of having transparent,
 public, collaborative, community standards for interoperability
 and I don't believe it was anyone's intention here.

### CODE REVIEW

**Comment by luke-jr** [comment]
I think a change like this really needs a new BIP.

**Comment by btcdrak** [comment]
ACK 3548ea23b624d0bd62f36b5bcd9cd08effca367c

**Comment by gmaxwell** [comment]
@luke-jr As of a week ago it would have pretty much been a total no-op.

Tricky here, so what we have a BIP-1 saying you can do X but then a BIP-N that says you can't? not exactly the best documentation. I suppose what we'd need to do is create a BIP-N that supersedes BIP1.  ... I suppose it's good that BIP1 isn't an immutable OPL document. :P

**Comment by kanzure** [comment]
ACK 3548ea23b624d0bd62f36b5bcd9cd08effca367c. My contributions to BIP1 should be considered part of the public domain.

I think there are some jurisdictions where authors cannot place works into the public domain directly. Perhaps some alternative recommendation should be made for those cases. Maybe the Creative Commons people figured out what to do about issues like this.

By the way, it would seem that we keep running into weird text found in BIP1. We should really encourage more people to read BIP1. I made BIP1 edits within the past year for some basic formatting issues such as fixing how entire new sections which were appended at the end of a line that nobody had noticed for years. This particular error had even been copied into other projects such as EIPs......

https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2016-September/013164.html

**Comment by btcdrak** [comment]
Creative Commons wrote about PD copyright here https://wiki.creativecommons.org/wiki/Public_domain

I also think we should avoid BIPs turning into a mirror for how legislation works where you need to scour several documents to find out the law. I think BIP1 should be amended, even if it's entirely replaced. BIP1 is clearly different to the other BIPs since it is the document that describes and defined the BIP process. It is already hard enough to get people to actually _read_ BIP1 without them having to read and merge several references.

**Comment by laanwj** [comment]
I think this makes sense.
ACK

File: bip-0001.mediawiki [hunk 1]
```diff
@@ -67,7 +67,7 @@ Each BIP should have the following parts:
 
 * Abstract -- a short (~200 word) description of the technical issue being addressed.
 
-* Copyright/public domain -- Each BIP must either be explicitly labelled as placed in the public domain (see this BIP as an example) or licensed under the Open Publication License.
+* Copyright/public domain -- Each BIP must either be explicitly labelled as placed in the public domain (see this BIP as an example).
```
Comment (MarcoFalke) [inline bip-0001.mediawiki]: This sentence no longer makes sense.

I'd suggest the following:

```mediawiki
* Copyright -- Each BIP must be explicitly labeled as placed in the public domain or licensed under at least one of the following licenses:
** BSD-2-Clause: [https://opensource.org/licenses/BSD-2-Clause OSI-approved BSD 2-clause license]
** CC0-1.0: [https://creativecommons.org/publicdomain/zero/1.0/ Creative Commons CC0 1.0 Universal]
** CC-BY-4.0: [https://creativecommons.org/licenses/by/4.0/ Creative Commons Attribution 4.0 International]
```

This would allow us to keep all the BIPs that are currently not placed in the public domain instead of having to delete them.

**Comment by MarcoFalke** [comment]
Concept ACK and I also agree with @btcdrak. BIP1 should always represent the current state of the BIP process/guidelines. I think it makes sense to incrementally improve BIP1 such that it converges to BIP2 over time. Otherwise we may run into the risk of not going anywhere, when it turns out that some people object specific changes of BIP2 and thus block any improvement that BIP2 tries to achieve.

File: bip-0001.mediawiki (hunk 1 above)
Comment (laanwj) [inline bip-0001.mediawiki]: A quick grep shows that the only BIPs with alternative licenses are:
- bip-0002:  dual-licensed under the Open Publication License and BSD 2-clause license
- bip-0145: dual-licensed under the Open Publication License and BSD 2-clause license

We should just try to get these re-licensed. I don't see anything under CC-BY-SA or CC0. Or I may have missed something?

BTW BIP1 **itself** also needs a copyright heading.

Comment (MarcoFalke) [inline bip-0001.mediawiki]: @laanwj Currently the copyright suggestions by BIP1 are a complete mess. I think we agree that OPL should be removed and at least BSD 2-clause be added, so we don't have to delete BIPs.

However, as we are touching this line anyway, we might want to provide an alternative for the public domain rule as well. As pointed out earlier, placing a work into the public domain is not possible (or even prohibited) in some countries. (c.f. http://opensource.stackexchange.com/a/1372)
Thus, choosing a license such as CC0, WTFPL, UNLINCENSE or similar is the legally "proper" way of placing a work in the public domain. To the best of my knowledge, CC0 seems to be the preferred way of placing something in the public domain.

Right now, adding CC0 to the acceptable licenses by BIP1 is a noop, but it would solve the issue in the future where a BIP author just wants to place the BIP into public domain and thus has to dual-license as BSD 2-clause and CC0, because just CC0 is not accepted by BIP1.

(I understand that my suggestion to add CC-BY-SA may be controversial, and I don't care too much, so we could just leave this for "later")

Comment (luke-jr) [inline bip-0001.mediawiki]: Would be helpful to put this effort into reviewing and finalising BIP 2...

Comment (laanwj) [inline bip-0001.mediawiki]: @MarcoFalke I agree.

@luke-jr Heh yes, this is typical scope creep. @gmaxwell 's point here is to get rid of a license that is clearly not a good idea for this kind of thing. Enumerating which licenses **are** a good match for the BIP process is important, but maybe for another time.

Comment (luke-jr) [inline bip-0001.mediawiki]: @laanwj We did the latter several months ago, but it got held up by arguments over what wiki to use for the (unrelated) BIP Comments. Might as well just finish it...

Comment (MarcoFalke) [inline bip-0001.mediawiki]: Though, I don't see what advantage there is to create another BIP to supersede BIP1, when it is possible to just amend BIP1. This will save a lot of confusion and time.

**Comment by super3** [comment]
@MarcoFalke Has there been any objections to BIP 2 in its current form? I can't find any.

**Comment by MarcoFalke** [comment]
No, I am not aware of any objections, which does not mean that no objections will come up.

**Comment by luke-jr** [comment]
BIP 1 has been Replaced by BIP 2, which no longer allows OPL. Considering this fixed.
